internal/distributed/querycoord/service.go

```diff
@@ -120,10 +124,21 @@ func (s *Server) init() error {
 	}
 	s.etcdCli = etcdCli
 	s.SetEtcdClient(etcdCli)
-	s.queryCoord.SetAddress(Params.GetAddress())
+	s.queryCoord.SetAddress(rpcParams.GetAddress())
+
+	if params.MetaStoreCfg.MetaStoreType.GetValue() == "tikv" {
+		log.Info("Connecting to tikv metadata storage.")
+		s.tikvCli, err = tikv.GetTiKVClient(&paramtable.Get().TiKVCfg)
+		if err != nil {
+			log.Debug("DataCoord failed to connect to tikv", zap.Error(err))
```

Comment:
Error message for all the Node types are set to datacoord

Reply:
nice catch